src/compiler/checker.ts

```diff
@@ -30954,7 +30954,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     // union type of return types from these signatures
     function getContextualSignature(node: FunctionExpression | ArrowFunction | MethodDeclaration): Signature | undefined {
         Debug.assert(node.kind !== SyntaxKind.MethodDeclaration || isObjectLiteralMethod(node));
-        const typeTagSignature = getSignatureOfTypeTag(node);
+        const typeTagSignature = getSignatureOfTypeTag(node, /*allowMembers*/ false);

```

Comment:
I came to a realization that this is the only call to `getSignatureOfTypeTag` that has to disallow members. However, I also find this call being here slightly off. A signature annotated through a type tag is still an annotated signature in my eyes and not a contextual signature. 

I think there is a chance that resolving this could be lifted and then `getSignatureOfTypeTag` could just **always** allow members. 

We could also allow members here too. It should error on the lack of those extra members but the param annotations could be taken into consideration.

Reply:
1. If this is the only place that disallows members, I do think it would be better to get the signature and then disallow them here.
2. I don't remember the situation precisely, I just remember that treating a type tag as context was a hack. It would be nice not to have it.

Reply:
1. I now allow members here. The "contextual" parameter types are assigned based on the type tag with members but it errors later because inline function expressions etc can't have members. I added a test case to this PR that verifies this behavior
2. Roget that. I might take a look at this sometime in the future but it's not relevant to this PR so I'd have to tackle it separately.